Clarify array of records (#3715)
modelica/ModelicaSpecification: chapters/functions.tex

The table below shows the mapping of an array argument in the absence of an explicit external function call when calling a C function.
The type {\lstinline!T!} is allowed to be any of the simple types which can be passed to C as defined in \cref{simple-types} or a record type as defined in \cref{records} and it is mapped to the type $T'$ as defined in these sections for input arguments.

> (Which for array of records mean passing a (possibly \lstinline[language=C]!const!) pointer to the first element of an array of structs in C.)

Reviewer: I don't think there should be any need for clarification here.  We should just be more clear in the section on records.
```suggestion
```

\item
  The components of the Modelica record class are declared in the same order in the C struct.
\item
  Records containing arrays cannot be mapped.
\end{itemize}

Scalar records are passed by reference (i.e., a pointer to the record is being passed).

> Array of records are passed as a pointer to the first element in an array of structs (usual array decay in C).

Reviewer: Just need to make clear that what is stated here is repeated information:
```suggestion
As stated in \cref{arrays-1}, an array of records is passed as a pointer to the first element in an array of structs (usual array decay in C).
```

> As stated in \cref{arrays-1}, an array of records is passed as a pointer to the first element in an array of structs (usual array decay in C).

Reviewer: Speeding up Dag's proposed change:
```suggestion
As stated in \cref{arrays-1}, an array of records is passed as a pointer to the first element in an array of structs (usual array-to-pointer decay in C).
```

\item
  The components of the Modelica record class are declared in the same order in the C struct.
\item
  Records containing arrays cannot be mapped.
\end{itemize}

> Records are passed by reference (i.e., a pointer to the record is being passed).

Reviewer: This is where I think clarification for arrays is needed.  Something like the following might be enough:
> Scalar records are passed by reference (i.e., a pointer to the record is being passed).
